### PR TITLE
python: Fixed parameter name to allow removal of LB boundaries.

### LINKED ISSUE #1935: LBBoundary can not be removed
``` python
import espressomd                                                       
import espressomd.lb                                                    
import espressomd.shapes                                                
import espressomd.lbboundaries                                          
                                                                        
system = espressomd.System(box_l=[10.0]*3, time_step = 0.01)            
system.cell_system.skin = 0.4                                           
                                                                        
lbf = espressomd.lb.LBFluidGPU(                                  
    visc=1.0,                                                           
    dens=1.0,                                                           
    agrid=1.0,                                                          
    tau=0.01,                                                           
    fric=1.0,                                                           
    ext_force = [0.0, 0.0, 0.1])                                        
system.actors.add(lbf)                                                                                      
wall_shape1 = espressomd.shapes.Wall(normal=[1., 0., 0.], dist=1.0)     
wall_shape2 = espressomd.shapes.Wall(normal=[-1., 0., 0.], dist=-9.0)   
boundary1 = espressomd.lbboundaries.LBBoundary(shape=wall_shape1)       
boundary2 = espressomd.lbboundaries.LBBoundary(shape=wall_shape2)       
system.lbboundaries.add(boundary1)                                      
system.lbboundaries.add(boundary2)                                      
for b in system.lbboundaries:                                                                                
    system.lbboundaries.remove(b)                                       
```

### PR DESCRIPTION
**Pull request by fweik**
Fixes #1935.

### CODE REVIEW

**Comment by RudolfWeeber** [comment]
Did you check that removing a boundary actually updates the boundary flags of the lb nodes?

**Comment by fweik** [comment]
It does not I think. Good point.

**Review by KaiSzuttor — APPROVED** [review]

**Comment by fweik** [comment]
Please don't merge, we still need to make sure the boundary field is updated is boundaries are added or removed.

**Comment by KaiSzuttor** [comment]
i checked the code and it seems not to be trivial to accomplish this.

**Comment by fweik** [comment]
Why can't we just call `lb_boundary_init` after every change?

**Comment by KaiSzuttor** [comment]
good point...

**Comment by fweik** [comment]
@KaiSzuttor Now of course the tests fail because the shear plan is shifted....

**Comment by fweik** [comment]
@KaiSzuttor ready for review. The refactoring was supposed to go to a different branch

**Comment by fweik** [comment]
Added basic test, apparently this is broken for the GPU version. Unfortunately the code is completely unreadable.

**Comment by KaiSzuttor** [comment]
feature guards missing in the script interface? feature guard in the testcase

File: testsuite/lb_boundary.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    def setUp(self):
+        self.system.actors.add(self.lbf)
+
+    def tearDown(self):
+        self.system.lbboundaries.clear()
+        self.system.actors.remove(self.lbf)
+
+@ut.skipIf(not espressomd.has_features(["LB_BOUNDARIES_GPU"]),
+           "Features not available, skipping test!")
+class LBBoundariesGPU(ut.TestCase, LBBoundariesBase):
```
Comment (KaiSzuttor) [inline testsuite/lb_boundary.py:104]: :+1:

File: testsuite/lb_boundary.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        lbf = self.lbf
+
+        for i in product(range(0, 5), rng, rng):
+            self.assertEqual(lbf[i].boundary, 1)
+
+        for i in product(range(5, 15), rng, rng):
+            self.assertEqual(lbf[i].boundary, 0)
+
+        for i in product(range(15, 20), rng, rng):
+            self.assertEqual(lbf[i].boundary, 2)
```
Comment (KaiSzuttor) [inline testsuite/lb_boundary.py:76]: How can this be two if its of type boolean in the core?

File: testsuite/lb_boundary.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        lbb.clear()
+        for i in product(rng, rng, rng):
+            self.assertEqual(lbf[i].boundary, 0)
+
+@ut.skipIf(not espressomd.has_features(["LB_BOUNDARIES"]),
+           "Features not available, skipping test!")
+class LBBoundariesCPU(ut.TestCase, LBBoundariesBase):
+    lbf = None
```
Comment (KaiSzuttor) [inline testsuite/lb_boundary.py:85]: this variable declaration plus the teardown method can go into the base

File: testsuite/lb_boundary.py (hunk 2 above)
Comment (fweik) [reply testsuite/lb_boundary.py:76]: Its not, it's an int (1 + the boundary id).

File: testsuite/lb_boundary.py (hunk 3 above)
Comment (fweik) [reply testsuite/lb_boundary.py:85]: I don't think it makes sense to further refactor the test while the core functionality is not working.

Comment (KaiSzuttor) [reply testsuite/lb_boundary.py:85]: this is orthogonal to the core functionality

Comment (fweik) [reply testsuite/lb_boundary.py:85]: Yes, but the test may change.

File: testsuite/lb_boundary.py (hunk 2 above)
Comment (KaiSzuttor) [reply testsuite/lb_boundary.py:76]: yes but in #2073 it's a bool... maybe we should merge those two PR's

Comment (fweik) [reply testsuite/lb_boundary.py:76]: No, this bugfix should go into 4.0, I don't think we have time to finish and test the other LB stuff until next week. I have still other things to do.

**Pull request merged**